Failing test due to keyring module upgrade

Originally detected in https://github.com/datalad/datalad/pull/2958#issuecomment-433878230
```
======================================================================
FAIL: datalad.downloaders.tests.test_credentials.test_credentials_from_env
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/downloaders/tests/test_credentials.py", line 108, in test_credentials_from_env
    assert_equal(cred.get('key_id'), '1')
AssertionError: None != '1'
-------------------- >> begin captured logging << --------------------
datalad: DEBUG: Importing keyring
datalad.downloaders.credentials: WARNING: Failed to query keyring: No recommended backend was available. Install the keyrings.alt package if you want to use the non-recommended backends. See README.rst for details. [fail.py:get_password:23]
--------------------- >> end captured logging << ---------------------
```
probably relates to the fresh keyring 16.0.0 instead of previous 15.1.0

--- a/datalad/support/external_versions.py
+++ b/datalad/support/external_versions.py
@@ -125,6 +125,8 @@ class ExternalVersions(object):
         'gitdb',
         'humanize',
         'iso8601',
+        'keyring',
+        'keyrings.alt',
         'msgpack',
         'mutagen',
         'patool',

--- a/datalad/support/keyring_.py
+++ b/datalad/support/keyring_.py
@@ -44,11 +44,11 @@ class Keyring(object):
 
     # proxy few methods of interest explicitly, to be rebound to the module's
     def get(self, name, field):
-        # consult environment, might be provided there
-        val = self._keyring.get_password(self._get_service_name(name), field)
-        if val is None:
-            val = os.environ.get(('DATALAD_%s_%s' % (name, field)).replace('-', '_'), None)
-        return val
+        # consult environment, might be provided there and should take precedence
+        env_var = ('DATALAD_%s_%s' % (name, field)).replace('-', '_')
+        if env_var in os.environ:
+            return os.environ[env_var]
+        return self._keyring.get_password(self._get_service_name(name), field)
 
     def set(self, name, field, value):
         return self._keyring.set_password(self._get_service_name(name), field, value)

